### PR TITLE
Use GetPassphrase everywhere (via delegate UI if available)

### PR DESCRIPTION
**Pull request by patrickxb**
This changes all use of the SecretUI to use GetPassphrase instead of the four existing functions.

The result of this is that if desktop app is running, the desktop app pinentry will be used instead of gpg pinentry (or terminal).

Note that after this gets approved, I will submit another PR that cleans up the SecretUI protocol so it just has GetPassphrase() and cleans up all the related code.  There is a lot of cruft now, but I wanted to get the version with no protocol changes in first.

r? @maxtaco 

cc: @chrisnojima @cjb

### CODE REVIEW

File: go/engine/login_provision.go [hunk 1]
```diff
@@ -756,13 +756,17 @@ func (e *LoginProvision) checkUserByPGPFingerprint(ctx *Context, fp *libkb.PGPFi
...
 	}
 	paperPhrase := libkb.NewPaperKeyPhrase(passphrase)
-	if paperPhrase.Version() != libkb.PaperKeyVersion {
-		e.G().Log.Debug("paper version mismatch:  generated paper key version = %d, libkb version = %d", paperPhrase.Version(), libkb.PaperKeyVersion)
+	version, err := paperPhrase.Version()
+	if err != nil {
+		return nil, err
+	}
+	if version != libkb.PaperKeyVersion {
+		e.G().Log.Debug("paper version mismatch:  generated paper key version = %d, libkb version = %d", version, libkb.PaperKeyVersion)
```
Comment (maxtaco) [inline go/engine/login_provision.go]: noooooooooooooo! :)

File: go/engine/common.go [hunk 2]
```diff
@@ -43,13 +42,17 @@ func findPaperKeys(ctx *Context, g *libkb.GlobalContext, me *libkb.User) (*keypa
...
 	}
 	paperPhrase := libkb.NewPaperKeyPhrase(passphrase)
-	if paperPhrase.Version() != libkb.PaperKeyVersion {
-		g.Log.Debug("paper version mismatch:  generated paper key version = %d, libkb version = %d", paperPhrase.Version(), libkb.PaperKeyVersion)
+	version, err := paperPhrase.Version()
+	if err != nil {
+		return nil, err
+	}
+	if version != libkb.PaperKeyVersion {
+		g.Log.Debug("paper version mismatch:  generated paper key version = %d, libkb version = %d", version, libkb.PaperKeyVersion)
```
Comment (maxtaco) [inline go/engine/common.go]: no spc after `:` pls

File: go/client/cmd_signup.go [hunk 3]
```diff
@@ -205,17 +205,13 @@ func (s *CmdSignup) prompt() (err error) {
...
-		UseSecretStore: libkb.HasSecretStore(),
-	}
 
 	f := s.fields.passphraseRetry
 	if f.Disabled || libkb.IsYes(f.GetValue()) {
+		// note that since this is running in the client, it is not
+		// possible to use a delegated SecretUI.
 		var res keybase1.GetPassphraseRes
-		res, err = s.G().UI.GetSecretUI().GetNewPassphrase(arg)
+		res, err = libkb.GetSignupPassphrase(s.G().UI.GetSecretUI())
```
Comment (maxtaco) [inline go/client/cmd_signup.go]: note that this code isn't only reachable from standalone mode. it's also reachable in client/server mode where the client is a CLI.

Comment (maxtaco) [inline go/client/cmd_signup.go]: insert TODO/ticket for a fix of this

File: go/engine/login_state_test.go [hunk 4]
```diff
@@ -78,43 +78,44 @@ func TestLoginLogout(t *testing.T) {
 // different from the main one, so don't mess with testing.T (which
 // isn't safe to use from a non-main goroutine) directly, and instead
 // have a LastErr field.
-type GetSecretMock struct {
+type GetPassphraseMock struct {
```
Comment (maxtaco) [inline go/engine/login_state_test.go]: can this class be non-exported?

Comment (patrickxb) [inline go/engine/login_state_test.go]: yes.

not a big deal, though, as it is only compiled in tests.

File: go/client/cmd_signup.go (hunk 3 above)
Comment (patrickxb) [inline go/client/cmd_signup.go]: The comment doesn't mention standalone mode.

File: go/libkb/keylock.go [hunk 5]
```diff
@@ -39,33 +35,25 @@ func (arg KeyUnlocker) Run() (ret GenericKey, err error) {
...
-		res, err = arg.UI.GetSecret(keybase1.SecretEntryArg{
-			Err:            emsg,
-			Desc:           desc,
-			Prompt:         prompt,
-			Reason:         arg.Reason,
-			UseSecretStore: arg.UseSecretStore,
-		}, nil)
-
-		if err == nil && res.Canceled {
-			err = CanceledError{"Attempt to unlock secret key entry canceled"}
```
Comment (maxtaco) [inline go/libkb/keylock.go]: is there no longer a relevant cancelation error?

**Comment by maxtaco** [comment]
Once remaining Q, otherwise :+1:

File: go/libkb/keylock.go (hunk 5 above)
Comment (patrickxb) [inline go/libkb/keylock.go]: electron pinentry returns InputCanceled when user clicks cancel.

**Pull request merged**